Add node coverage.

Kumar-Tarun opened the pull request:
This PR currently adds coverage for `hash_util.js`. I also plan to add coverage for some more files. So, please don't close this PR (I requested this in the previous PR too). :)

showell commented:
Looks good so far.  I'm a little way of the variables `result2`, `result3`, etc.  I get the concept that re-using `result` over and over again is a lot of mutation, but I think it just reads nicer and rarely creates practical problems.  And it's possible here a tiny 3-line helper function can clean this up a bit.

showell commented:
@timabbott I don’t understand your comment, this is two small commits that aren’t likely to cause problems with the bigger efforts.

showell commented:
Oh I think he just means don’t close the PR?  I agree we should merge once the comments are addressed.

showell commented:
@Kumar-Tarun I'll try to explain the best way to proceed here:

If it's the case that the first two commits are ready to merge, then you should expect us to merge them, and if as a side effect of merging, we close the PR, there's nothing to prevent you from just opening a new PR.  That's kind of the normal workflow.

If for some reason the first two commits aren't ready, then you should put "wip" in the title of the PR.  And I don't really recommend that strategy.  As Tim says, and I'm paraphrasing, it's generally a good strategy to get small PRs merged before you build up large PRs that are hard to review and can build up conflicts with other ongoing work.

Does that make sense?

Kumar-Tarun commented:
@showell sorry for the confusion here. My comment means that currently I have added some files for node coverage, and plan to add more in future, so I just wanted that this PR not be closed, so that I can submit commits related to node coverage under one PR, and the commits which have been reviewed and updated get merged. I just thought that why should I again and again open a new PR for the same purpose. Thanks for the review, I will update soon! :)

Kumar-Tarun commented:
@showell I've update the PR, you may check!
